montecarlo_localization.py

```diff
@@ -326,13 +327,16 @@ def sample_motion(self, msg):
         self.prev_log_pose = msg_pose # Save previous log pose for delta
         return self.pose
 
-    def new_pose_from_sample_error(self, scale=10):
+    def new_pose_from_sample_error(self, scale=10, constrainedPose=[0,0,0]):
```

Comment:
Just came to fix this and noticed a much worse bad than the case thing.. Using a mutable value as a default argument is a really common gotcha - https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments